src/module_wrap.h

```diff
@@ -12,6 +12,12 @@
 namespace node {
 namespace loader {
 
+struct PackageConfig {
+  bool exists;
+  bool has_main;
+  std::string main;
```

Comment:
Make these const if at all possible.

Reply:
I couldn't get this to work unfortunately - seems to be a result of `PackageConfig` not being a pointer on the map itself in the assignment operator.

Reply:
What was the error you got?

Reply:
These two:

```
error: use of deleted function ‘node::Environment::PackageConfig& node::Environment::PackageConfig::operator=(const node::Environment::PackageConfig&)’

error: non-static const member ‘const bool node::Environment::PackageConfig::exists’, can’t use default assignment operator
```

Reply:
@guybedford that's just because since it's const it must be fully intiialized - when you do `myMap['foo'] = bar` that potentially creates an empty object and then assigns `bar` to it.

You can use `emplace` to construct the object inside the map to overcome it.